src/app/AttributeCache.cpp

```diff
@@ -219,5 +219,75 @@ CHIP_ERROR AttributeCache::GetStatus(const ConcreteAttributePath & path, StatusI
     return CHIP_NO_ERROR;
 }
 
+struct compare
+{
+    bool operator()(const ConcreteDataAttributePathWithSize & x, const ConcreteDataAttributePathWithSize & y) const
+    {
+        return x.mSize < y.mSize;
+    }
+};
+
+uint32_t AttributeCache::OnUpdateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                       const Span<DataVersionFilter> & aDataVersionFilters)
+{
+    uint32_t number = 0;
+    CHIP_ERROR err  = CHIP_NO_ERROR;
+    TLV::TLVWriter backup;
+
+    std::set<ConcreteDataAttributePathWithSize, compare> clusterSet;
```

Comment:
I don't think `std::set` is the most appropriate choice here. That effectively means you're forced to mash together the logical key (which is the path) and the value (which is the size) together into this composited struct. 

This struct is also 'leaky' in that it should really not exist anywhere beyond the scope of this function (it looks like it was added to ConcreteAttributePath.h, which it shouldn't).

I'd rather it be a `std::map<ConcreteAttributePath, size_t>` with a custom comparator.

Reply:
Done, thanks